test/e2e/create_cluster_test.go

```diff
@@ -217,24 +182,32 @@ func TestCreateClusterProxy(t *testing.T) {
 	numNodes := clusterOpts.NodePoolReplicas * int32(len(clusterOpts.AWSPlatform.Zones))
 	e2eutil.WaitForNReadyNodes(t, testContext, guestClient, numNodes, hostedCluster.Spec.Platform.Type)
 
-	// TODO (alberto): move into WaitForNReadyNodes after this PR github.com/openshift/hypershift/pull/1702 gets merged so it's validated by any call to the function in any test.
-	// It's has to wait for the PR to merged otherwise the control_plane_upgrade_test would fail.
-	t.Logf("Validating all Nodes have the NodePool label")
-	nodes := &corev1.NodeList{}
-	if err := guestClient.List(ctx, nodes); err != nil {
-		t.Fatalf("failed to list nodes in guest cluster: %v", err)
-	}
-	for _, node := range nodes.Items {
-		g.Expect(node.Labels[hyperv1.NodePoolLabel]).NotTo(BeEmpty())
-	}
-
 	// Wait for the rollout to be complete
 	t.Logf("Waiting for cluster rollout. Image: %s", globalOpts.LatestReleaseImage)
 	e2eutil.WaitForImageRollout(t, testContext, client, hostedCluster, globalOpts.LatestReleaseImage)
-	err = client.Get(testContext, crclient.ObjectKeyFromObject(hostedCluster), hostedCluster)
+
+	err := client.Get(testContext, crclient.ObjectKeyFromObject(hostedCluster), hostedCluster)
 	g.Expect(err).NotTo(HaveOccurred(), "failed to get hostedcluster")
 
-	e2eutil.EnsureNodeCountMatchesNodePoolReplicas(t, testContext, client, guestClient, hostedCluster.Namespace)
+	e2eutil.EnsureNodeCountMatchesNodePoolReplicas(t, ctx, client, guestClient, hostedCluster.Namespace)
 	e2eutil.EnsureNoCrashingPods(t, ctx, client, hostedCluster)
 	e2eutil.EnsureNodeCommunication(t, ctx, client, hostedCluster)
 }
+
+func validatePrivateCluster(t *testing.T, ctx context.Context, client crclient.Client, hostedCluster *hyperv1.HostedCluster) {
+	g := NewWithT(t)
```

Comment:
can we please include here validations for known expectations:
E.g. the format of the hc.status.controlplaneendpoint is as expected then "public" URL and not local?